ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -76,8 +77,11 @@ class MSFButtonStateImpl: NSObject, ObservableObject, ControlConfiguration, MSFB
     @Published var style: MSFButtonStyle
 
     @Published var overrideTokens: ButtonTokens?
-    @Published var tokens: ButtonTokens
-    var defaultTokens: ButtonTokens { .init(style: self.style, size: self.size) }
+    @Published var tokens: ButtonTokens = .init() {
```

Comment:
Have you tested switching the style in other way than the video in the description?

The reason I bring it up is because tapping the button causes the SwiftUI refresh as it needs to render state changes like the pressed state. 

It may not be updating because of the tokens/style changes, but because of the action of tapping it.

It's something worth validating just to make sure.

Reply:
Great point! I validated with a fun new bit of code in the `ButtonDemoController`:

```swift
    private func dequeueDemoButton(indexPath: IndexPath,
                                   style: MSFButtonStyle,
                                   size: MSFButtonSize,
                                   disabled: Bool) -> MSFButton {
        let key = "\(indexPath)-\(disabled)"
        if let button = buttons[key] {
            return button
        } else {
            let button = MSFButton(style: style,
                                   size: size) { [weak self] _ in
                self?.buttons.values.forEach { aButton in
                    aButton.state.style = (aButton.state.style == .secondary) ? .primary : .secondary
                }
                self?.didPressButton()
            }
            button.state.isDisabled = disabled
            button.state.overrideTokens = self.overrideTokens
            buttons[key] = button

            return button
        }
    }
```

And here's the successful result:

https://user-images.githubusercontent.com/4934719/152887018-aa1e753c-8197-4fae-8d67-3074e91bba0e.mp4